plugins/woocommerce/src/StoreApi/Authentication.php

```diff
@@ -139,6 +141,34 @@ public function set_logged_in_cookie( $logged_in_cookie ) {
 		$_COOKIE[ LOGGED_IN_COOKIE ] = $logged_in_cookie;
 	}
 
+	/**
+	 * Opt in to rate limiting for the checkout endpoint.
+	 *
+	 * @param \WP_Error|mixed $result Error from another authentication handler, null if we should handle it, or another value if not.
+	 * @return \WP_Error|null|bool
+	 */
+	public function opt_in_checkout_endpoint( $result ) {
+		if (
+			FeaturesUtil::feature_is_enabled( 'rate_limit_checkout' )
+			&& $this->is_request_to_store_api()
+			&& preg_match( '#/wc/store(?:/v\d+)?/checkout#', $GLOBALS['wp']->query_vars['rest_route'] )
+			&& isset( $_SERVER['REQUEST_METHOD'] )
+			&& 'POST' === $_SERVER['REQUEST_METHOD']
+		) {
+			add_filter(
+				'woocommerce_store_api_rate_limit_options',
+				function ( $options ) {
+					$options['enabled'] = true;
+					$options['limit']   = 3;
+					$options['seconds'] = 60;
+					return $options;
```

Comment:
Will this overwrite values from the plugin? should we be fetching this from options, with those defaults?

```suggestion
                    return array(
                        'enabled' => true, // Overriden just for /checkout
                        'limit' => get_option( 'rate_limiting_limit', 3 ),
                        'seconds' => get_option( 'rate_limiting_seconds', 60 ),
                        'proxy_support' => get_option( 'rate_limiting_proxy_support', 'no' ),
                    );
```

Reply:
I don't think we can really recommend that plugin without updating it, the options aren't prefixed. This can cause issues in the future.

Reply:
> Will this overwrite values from the plugin?

It depends at which priority the plugin is hooking to the filter. I don't think we should be using options, I'm hooking into the filter that comes with the rate limiter.

> I don't think we can really recommend that plugin without updating it, the options aren't prefixed. This can cause issues in the future.

We have plenty of time to update it or remove it. The docs sits on our repo, and not released. Users will only see this once 9.6 is out next year.

Reply:
>It depends at which priority the plugin is hooking to the filter. I don't think we should be using options, I'm hooking into the filter that comes with the rate limiter.

Sorry confused tabs and missed the priority arg.